### PR TITLE
[OF-1473] feat: Change URL to Multiplayer Service

### PR DESCRIPTION
**Pull request by MAG-OR**
- Update domain name for Multiplayer Service in CSP initialisation
- Update WebSocketClientTests to use the new hostname

### CODE REVIEW

File: Tests/src/InternalTests/WebSocketClientTests.cpp [hunk 1]
```diff
@@ -40,7 +40,7 @@ CSP_INTERNAL_TEST(CSPEngine, WebSocketClientTests, SignalRClientStartStopTest)
 	LogIn(UserSystem, UserId);
 
 	// Start
-	auto* WebSocket = WebSocketStart("wss://ogs-odev-internal.magnoboard.com/mag-multiplayer/hubs/v1/multiplayer?id=");
+	auto* WebSocket = WebSocketStart("wss://ogs-multiplayer-odev-internal.magnoboard.com/mag-multiplayer/hubs/v1/multiplayer?id=");
```
Comment (jeromagnopus) [inline Tests/src/InternalTests/WebSocketClientTests.cpp:43]: nitpick: I know they're just tests, but I would still argue against any hardcoded URLs in code.

File: Library/src/CSPFoundation.cpp [hunk 2]
```diff
@@ -255,10 +255,15 @@ bool CSPFoundation::Initialise(const csp::common::String& EndpointRootURI, const
...
 	const std::string PrototypeServiceURI	= RootURI + "/mag-prototype";
 	const std::string SpatialDataServiceURI = RootURI + "/mag-spatialdata";
-	const std::string MultiplayerServiceURI = RootURI + "/mag-multiplayer/hubs/v1/multiplayer";
 	const std::string AggregationServiceURI = RootURI + "/oly-aggregation";
 	const std::string TrackingServiceURI	= RootURI + "/mag-tracking";
 
+	const std::string Hostname = "ogs", MultiplayerHostname = Hostname + "-multiplayer";
+	const std::string RootMultiplayerURI
+		= RootURI.substr(0, RootURI.find(Hostname)) + MultiplayerHostname + RootURI.substr(RootURI.find(Hostname) + Hostname.length());
+	const std::string MultiplayerServiceURI = RootMultiplayerURI + "/mag-multiplayer/hubs/v1/multiplayer";
```
Comment (MAG-ElliotMorris) [inline Library/src/CSPFoundation.cpp:264]: [Consider] It's pretty hard to follow what this is doing without understanding what the intended constructed string is, a comment or two describing the string transformation could help.
I wonder if it's necessary to go to the effort to transform this from the `RootURI`, could we not directly encode the new multiplayer URI and void some of this transformation?

**Review by MAG-ElliotMorris — APPROVED** [review]
Approved with optional comment.

File: Library/src/CSPFoundation.cpp (hunk 2 above)
Comment (MAG-OR) [reply Library/src/CSPFoundation.cpp:264]: Sure I'll add comments. 

I constructed it from RootURI because it is not always the same. It is passed as an argument in the Foundation initialisation. The production one is ogs.magnopus-something, but there are also others like ogs-internal.magnopus-something. 
I made the assumption that ogs-multiplayer-internal was the right order, thinking if it wasn't the CI would tell me, but now I am not so sure we have the test coverage for that.

I reached out to CHS and the person to ask was off yesterday, back on Monday

Comment (MAG-ElliotMorris) [reply Library/src/CSPFoundation.cpp:264]: Ah right, sorry, I had thought this was in the tests, not the core lib itself.
Aye ... it is rather difficult to be certain that this is fully correct. I couldn't say for sure if this URI construction was always right, or is just right in the case of the particular services CHS has stood up right now.

Comment (MAG-OR) [reply Library/src/CSPFoundation.cpp:264]: Agreed, I'll ask that person on Monday for guidance

Comment (MAG-OR) [reply Library/src/CSPFoundation.cpp]: Comment request fixed in 18eb922b567bae6e56fc48f7b657eb7cf5257a3d

File: Tests/src/InternalTests/WebSocketClientTests.cpp (hunk 1 above)
Comment (raynopus) [reply Tests/src/InternalTests/WebSocketClientTests.cpp:43]: This shouldn't work. The URL is pointing to the old Dev environment which doesn't exist anymore. Can we take this opportunity to change the URL to `wss://ogs-multiplayer-internal.magnopus-dev.cloud/mag-multiplayer/hubs/v1/multiplayer`?

Comment (MAG-OR) [reply Tests/src/InternalTests/WebSocketClientTests.cpp:43]: Sure thing. Does "-multiplayer" always come before other strings like "-internal"?

Comment (MAG-OR) [reply Tests/src/InternalTests/WebSocketClientTests.cpp]: awesome, thank you for confirming

**Review by MAG-ElliotMorris — APPROVED** [review]

**Review by MAG-ChristopherAtkinson — APPROVED** [review]

**Review by MAG-mv — APPROVED** [review]
